### PR TITLE
RFC: Set to Set conversions

### PR DESCRIPTION
**Pull request by ghost**
As a major set fan I found it somewhat odd that `convert(Vector{String}, split("foo bar baz qux"))` works, but the same logic for sets, `convert(Set{String}, Set(split("foo bar baz qux")))`, doesn't.  This is a patch based on my own little hack that I introduced in a module to get around the aforementioned issue.  Feedback is as always most welcome.

### CODE REVIEW

**Comment by ninjin** [comment]
Test failure on x86_64 appears unrelated, should I force push to trigger a re-run?

**Comment by vchuravy** [comment]
Travis log backed-up here: https://gist.github.com/vchuravy/b3e833ed16d6f7bde9aa65c039b3a2dc

File: base/set.jl [hunk 1]
```diff
@@ -193,3 +193,6 @@ function hash(s::Set, h::UInt)
     end
     return h
 end
+
+convert{T}(::Type{Set{T}}, s::Set{T}) = s
+convert{T,S}(::Type{Set{T}}, x::Set{S}) = Set{T}(convert(T, e) for e in x)
```
Comment (JeffBezanson) [inline base/set.jl]: I believe just `Set{T}(x)` would work.

**Review by JeffBezanson — CHANGES_REQUESTED** [review]

**Comment by JeffBezanson** [comment]
:+1: Really surprising we didn't have this already!

File: base/set.jl (hunk 1 above)
Comment (ninjin) [inline base/set.jl]: Thank you, I really should have realised that one on my own.

**Comment by ninjin** [comment]
Improved version pushed.  I also think I will submit a similar PR for `Pair`.

File: test/sets.jl [hunk 2]
```diff
@@ -252,3 +252,13 @@ end
 @test pop!(Set(1:2), 2, nothing) == 2
 
 @test length(Set(['x',120])) == 2
+
+# convert
+let
+    iset = Set([17, 4711])
+    @test convert(Set{Float64}, iset) == iset
+    fset = Set([17.0, 4711.0])
+    @test convert(Set{Int}, fset) == fset
```
Comment (tkelman) [inline test/sets.jl]: test the element types of the result if possible

Comment (ninjin) [inline test/sets.jl]: In a hurry this morning, so I only dug through the manual briefly and pushed the cleanest (still dirty) way I could think of doing this.  More julian suggestions are most welcome.

**Review by JeffBezanson — APPROVED** [review]

**Pull request merged**